applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.hpp

```diff
@@ -314,6 +313,11 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) SmallStrainUPwDiffOrderElement : pub
 
     Vector GetPressureSolutionVector();
 
+    std::vector<double> CalculateDegreesOfSaturation(const std::vector<double>& fluid_pressures,
+                                                     RetentionLaw::Parameters& RetentionParameters);
+    std::vector<double> CalculateDerivativesOfSaturation(const std::vector<double>& fluid_pressures,
+                                                         RetentionLaw::Parameters& RetentionParameters);
```

Comment:
Is there any reason why these member functions can't be `const`? And perhaps you may want to add attribute `[[nodiscard]]` to them?